app/celery/process_sns_receipts_tasks.py

```diff
@@ -58,7 +57,7 @@ def process_sns_results(self, response):
         notifications_dao._update_notification_status(
             notification=notification,
             status=notification_status,
-            provider_response=provider_response if notification_status == NOTIFICATION_TECHNICAL_FAILURE else None,
+            provider_response=provider_response if notification_status else None,
```

Comment:
Yeah it's against the idea of the changes. I think ideally, we'd want to not override the previous value if current one is None (and if called before).. but it might be more difficult than what it's worth it and I'd always expect a provider response.

Let's just do this then?

```suggestion
            provider_response=provider_response,
```